selfdrive/athena/athenad.py

```diff
@@ -244,6 +254,80 @@ def b64jpeg(x):
     raise Exception("not available while camerad is started")
 
 
+def get_logs_to_send_sorted(curr_time):
+  # TODO: scan once then use inotify to detect file creation/deletion
+  logs = []
+  for log_entry in os.listdir(SWAGLOG_DIR):
+    log_path = os.path.join(SWAGLOG_DIR, log_entry)
+    try:
+      time_sent = int.from_bytes(getxattr(log_path, LOG_ATTR_NAME), sys.byteorder)
+    except (ValueError, TypeError):
+      time_sent = 0
+    # assume send failed and we lost the response if sent more than one hour ago
+    if not time_sent or curr_time - time_sent > 3600:
+      logs.append(log_entry)
+  # return logs in order they should be sent
+  # excluding most recent (active) log file
+  return sorted(logs[:-1])
+
+
+def log_handler(end_event):
+  if PC:
+    return
+
+  log_files = []
+  last_scan = 0
+  log_retries = 0
+  while not end_event.is_set():
+    try:
+      try:
+        result = json.loads(log_recv_queue.get(timeout=1))
+        log_success = result.get("success")
+        log_entry = result.get("id")
+        log_path = os.path.join(SWAGLOG_DIR, log_entry)
+        if log_entry and log_success:
+          try:
+            setxattr(log_path, LOG_ATTR_NAME, LOG_ATTR_VALUE_MAX_UNIX_TIME)
+          except OSError:
+            pass # file could be deleted by log rotation
+      except queue.Empty:
+        pass
+
+      curr_time = int(time.time())
```

Comment:
use `sec_since_boot()`, see comment above.

Reply:
yeah, I missed this one, thanks!